### ISSUE TITLE
Add MetadataColumn[Time] type to QIIME 2

### ISSUE THREAD

**Issue by ebolyen** [caporaso-lab/q2-covid-19#10]
We are getting by with Categorical in the plugin, but this feature would be pretty handy overall.

**Comment by nbokulich** [comment]
Yes! Maybe this should go in q2-types? I can imagine q2-longitudinal or other plugins could make use of such a type.

**Comment by ebolyen** [comment]
Technically it would be a framework change, since this is a new primitive, but I just put it in this repo as it seems to be holding the "bag of work" we need to do sooner rather than later.
